sessiontxn/isolation/optimistic.go

```diff
@@ -44,6 +50,45 @@ func NewOptimisticTxnContextProvider(sctx sessionctx.Context, causalConsistencyO
 	return provider
 }
 
+// IsTxnRetryable (if returns true) means the transaction could retry.
+// We only consider retry in this optimistic mode.
+// If the session is already in transaction, enable retry or internal SQL could retry.
+// If not, the transaction could always retry, because it should be auto committed transaction.
+// Anyway the retry limit is 0, the transaction could not retry.
+func isTxnRetryable(sessVars *variable.SessionVars, tp *sessiontxn.EnterNewTxnType) bool {
```

Comment:
isTxnRetryable => isOptimisticTxnRetryable may be better because it is only used by optimistic mode.

Reply:
Done.